lib/test-result-formatting.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+var chalk = require('chalk');
+var _ = require('underscore'); // TODO: `lodash` is more performant and more widely used these days
+
+var formatSuccessLine = function(success) {
+  return success['name'] + '#' + success['methodName'] + ' took ' + success['time'];
+};
+
+var formatSuccesses = function(testResult) {
+  var successes = testResult['successes'];
+
+  if (!successes || !successes.length) { return []; }
+
+  return ['Successes:'].concat(_.map(
+    successes, _.compose(chalk.green, formatSuccessLine)
+  ));
+};
+
+var formatFailure = function(failure) {
+  return [
+    chalk.red(failure['name'] + '#' + failure['methodName'] + ' took ' + failure['time']),
+    '  - ' + failure['message'],
+    '  - ' + failure['stackTrace']
+  ];
+};
+
+var formatFailures = function(testResult) {
+  var failures = testResult['failures'];
+
+  if (!failures || !failures.length) { return []; }
+
+  return _.flatten(['Failures:'].concat(_.map(failures, function (failure) {
+    return formatFailure(failure);
+  })));
+};
+
+var formatOverview = function(testResult) {
+  var successCount = (testResult['successes'] && testResult['successes'].length) || 0;
```

Comment:
mind that testResults['successes'] might be an object instead of an array.

So maybe s.th. like this:
```javascript
var successCount = ensureArray(testResult['successes']).length;
```